src/3rd_party_remove.c

```diff
@@ -146,10 +167,29 @@ enum swupd_code third_party_remove_main(int argc, char **argv)
 	ret = mom_get_manifests_list(mom, &mom->submanifests, is_installed_bundle_data);
 	if (ret) {
 		ret = SWUPD_COULDNT_LOAD_MANIFEST;
-		goto exit;
+		goto remove_repo;
 	}
 	mom->files = consolidate_files_from_bundles(mom->submanifests);
 
+	/* if there are errors at this point the repo may be corrupt,
+	 * continue only if forced by the user */
+remove_repo:
+	if (ret) {
+		if (cmdline_option_force) {
+			char *bin_dir;
+			bin_dir = third_party_get_bin_dir();
+			warn("The --force option is specified; forcing the removal of the repository\n");
+			info("Please be aware that this may leave dangling files in %s\n", bin_dir);
+			free_and_clear_pointer(&bin_dir);
+			ret_partial = ret;
+			ret = SWUPD_OK;
+		} else {
+			error("The 3rd-party repository %s is corrupt and cannot be removed cleanly\n", name);
```

Comment:
I think it's better to write something here about the possible dangling files too. Something like:
```
The 3rd-party repository %s is corrupt and cannot be removed cleanly
Some exported files can be left behind at /opt/3rd-party/bin/
```

Reply:
sounds good, I'll make the change